packages/repository/src/relations/has-many/has-many-through.repository.ts

```diff
@@ -238,21 +228,38 @@ export class DefaultHasManyThroughRepository<
   }
 
   async link(
-    targetModelId: TargetID,
+    targetInstance: TargetEntity,
```

Comment:
I find the requirement to provide the entire target entity as suboptimal.

Consider the following REST API call intending to create a link between Category 123 and Product 569.

```
POST /api/categories/123/products/link/569
```

I believe our current design does not require the controller to make any extra database calls before it can create the link. IIRC:

```ts
await this.categoryRepository.products(123).link(569);
```

In your new design, the controller would have to talk to the database first. I find this suboptimal performance-wise.

Can we revert the API change back to link/unlink accepting an id only?

Reply:
Code changes in: https://github.com/strongloop/loopback-next/pull/5719/commits/c499b5b3eb83f5aa2bccd0093ab6bbf3710c0efc and https://github.com/strongloop/loopback-next/pull/5719/commits/f23f2d6195eb50f9871327bb13d03f55d0ee1365.